packages/typography/src/Typography.tsx

```diff
@@ -136,4 +137,138 @@ const Overline: ExtendableBox<{
   return <Box className={cx(sharedStyles, overline, className)} {...rest} />;
 };
 
-export { H1, H2, Subtitle, Body, InlineCode, Disclaimer, Overline };
+const OpenInNewTab = (props: JSX.IntrinsicElements['svg']) => {
+  return (
+    <svg
+      width="16px"
+      height="16px"
+      viewBox="0 0 16 16"
+      xmlns="http://www.w3.org/2000/svg"
+      xmlnsXlink="http://www.w3.org/1999/xlink"
+      {...props}
+    >
+      <title>Open in New Tab</title>
+      <g stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
+        <path
+          d="M9.29411765,3 L10.7727647,4.47864706 L6.42788235,8.82352941 L7.17647059,9.57211765 L11.5218824,5.22776471 L13,6.70588235 L13,3 L9.29411765,3 Z M4,4.58823529 L4,12 L11.4122941,12 L11.4122941,7.76470588 L10.3529412,7.76470588 L10.3529412,10.9411765 L5.05882353,10.9411765 L5.05882353,5.64705882 L8.23529412,5.64705882 L8.23529412,4.58823529 L4,4.58823529 Z"
+          fill="#007CAD"
+        />
+      </g>
+    </svg>
+  );
+};
+
+const anchorDataProp = createDataProp('anchor-container');
+
+const linkStyles = css`
+  text-decoration: none;
+  color: ${uiColors.blue.base};
+  cursor: pointer;
+
+  span {
+    position: relative;
+
+    &:after {
+      content: '';
+      position: absolute;
+      top: calc(100% + 2px);
+      left: 0;
+      right: 0;
+      opacity: 0;
+      transform: scale(0.8, 1);
+      transition: 150ms ease-in-out;
+      height: 3px;
+      border-radius: 50px;
+    }
+  }
+
+  &:hover {
+    span:after {
+      opacity: 1;
+      transform: scale(1);
+      background-color: ${uiColors.gray.light2};
+    }
+  }
+
+  &:focus {
+    outline: none;
+
+    span:after {
+      opacity: 1;
+      transform: scale(1);
+      background-color: #9dd0e7;
+    }
+  }
+`;
+
+const arrowRightIcon = css`
+  height: 10px;
+  width: 10px;
+  opacity: 0;
+  transform: translate3d(-3px, 0, 0px);
+  transition: all 100ms ease-in;
+  ${anchorDataProp.selector}:hover &,
+  ${anchorDataProp.selector}:hover & {
+    opacity: 1;
+    transform: translate3d(3px, 0, 0px);
+  }
+`;
+
+type LinkProps = JSX.IntrinsicElements['a'] & {
+  showArrow?: boolean;
+};
+
+const Link = ({
+  href,
+  children,
+  className,
+  target: targetProp,
+  showArrow = false,
+  ...rest
+}: LinkProps) => {
+  if (!href) {
+    console.error(
+      'Link components are wrapped anchor tags. Please provide a destination URL through the `href` prop or consider using another component.',
+    );
+    return null;
+  }
+
+  const size = useBaseFontSize();
+  const fontSize = size === 16 ? typeScale2 : typeScale1;
+
+  const hrefHostname = new URL(href).hostname;
+  const currentHostname = window.location.hostname;
+  let target;
+
+  if (targetProp) {
+    target = targetProp;
+  } else {
+    if (hrefHostname === currentHostname) {
+      target = '_self';
+    } else {
+      target = '_blank';
+    }
+  }
+
+  const icon =
+    target === '_blank' ? (
+      <OpenInNewTab />
+    ) : showArrow ? (
```

Comment:
it might be confusing that `showArrow` being set to `true` doesn't guarantee that the arrow is shown. is there another name we could use to more clearly indicate that, e.g. `canShowArrow`

Reply:
yes good call

Reply:
the design specs say that the external icon should override this one